### ISSUE TITLE
external storage: NTFS errors apparently related to copying files

### ISSUE THREAD

**Issue by ben-en** [Outernet-Project/lighthouse-firmware#7]
As the title says, i believe these errors are related to copying files from internal storage to external storage. There also may be some hangups with the config of paths, or FSAL.
## Plugin behavior

Status lights blink quickly then turn solid

In syslog the following was observed:
#### NTFS - eHDD

```
Nov  4 19:57:51 outernet user.warn kernel: [ 1120.752216@1] Indeed it is in host mode hprt0 = 00021501
Nov  4 19:57:51 outernet user.info kernel: [ 1120.937371@1] usb 2-1: new high-speed USB device number 3 using dwc_otg
Nov  4 19:57:51 outernet user.warn kernel: [ 1120.938224@0] Indeed it is in host mode hprt0 = 00001101
Nov  4 19:57:51 outernet user.info kernel: [ 1121.161351@0] usb 2-1: New USB device found, idVendor=1058, idProduct=0820
Nov  4 19:57:51 outernet user.info kernel: [ 1121.161536@0] usb 2-1: New USB device strings: Mfr=1, Product=2, SerialNumber=5
Nov  4 19:57:51 outernet user.info kernel: [ 1121.161703@0] usb 2-1: Product: My Passport 0820
Nov  4 19:57:51 outernet user.info kernel: [ 1121.161828@0] usb 2-1: Manufacturer: Western Digital
Nov  4 19:57:51 outernet user.info kernel: [ 1121.161951@0] usb 2-1: SerialNumber: 575846314538344135333057
Nov  4 19:57:51 outernet user.info kernel: [ 1121.169562@0] usb-storage 2-1:1.0: USB Mass Storage device detected
Nov  4 19:57:51 outernet user.info kernel: [ 1121.171909@0] scsi1 : usb-storage 2-1:1.0
Nov  4 19:57:52 outernet user.notice kernel: [ 1122.171436@0] scsi 1:0:0:0: Direct-Access     WD       My Passport 0820 1012 PQ: 0 ANSI: 6
Nov  4 19:57:52 outernet user.notice kernel: [ 1122.194818@0] sd 1:0:0:0: [sda] Spinning up disk...
Nov  4 19:57:52 outernet user.notice kernel: [ 1122.206882@0] scsi 1:0:0:1: Enclosure         WD       SES Device       1012 PQ: 0 ANSI: 6
Nov  4 19:57:58 outernet user.warn kernel: [ 1123.207422@0] ......ready
Nov  4 19:57:58 outernet user.notice kernel: [ 1128.270591@0] sd 1:0:0:0: [sda] 1953458176 512-byte logical blocks: (1.00 TB/931 GiB)
Nov  4 19:57:58 outernet user.notice kernel: [ 1128.272560@0] sd 1:0:0:0: [sda] Write Protect is off
Nov  4 19:57:58 outernet user.debug kernel: [ 1128.272767@0] sd 1:0:0:0: [sda] Mode Sense: 47 00 10 08
Nov  4 19:57:58 outernet user.err kernel: [ 1128.274622@0] sd 1:0:0:0: [sda] No Caching mode page found
Nov  4 19:57:58 outernet user.err kernel: [ 1128.274771@0] sd 1:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:57:58 outernet user.err kernel: [ 1128.291029@1] sd 1:0:0:0: [sda] No Caching mode page found
Nov  4 19:57:58 outernet user.err kernel: [ 1128.291180@1] sd 1:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:57:58 outernet user.info kernel: [ 1128.353259@1]  sda: sda1
Nov  4 19:57:58 outernet user.err kernel: [ 1128.369156@1] sd 1:0:0:0: [sda] No Caching mode page found
Nov  4 19:57:58 outernet user.err kernel: [ 1128.369303@1] sd 1:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:57:58 outernet user.notice kernel: [ 1128.369422@1] sd 1:0:0:0: [sda] Attached SCSI disk
Nov  4 19:57:59 outernet user.warn hotplug.sda1: Checking this filesystem is not supported
Nov  4 19:57:59 outernet user.debug hotplug.sda1: Mounting /dev/sda1 -> /mnt/sda1
Nov  4 19:57:59 outernet user.info hotplug.sda1: /dev/sda1 mounted at /mnt/sda1
Nov  4 19:57:59 outernet daemon.notice ntfs-3g[1354]: Version 2015.3.14 integrated FUSE 27
Nov  4 19:57:59 outernet daemon.notice ntfs-3g[1354]: Mounted /dev/sda1 (Read-Write, label "EXT_STORE", NTFS 3.1)
Nov  4 19:57:59 outernet daemon.notice ntfs-3g[1354]: Cmdline options: windows_names,fmask=133,dmask=022,recover
Nov  4 19:57:59 outernet daemon.notice ntfs-3g[1354]: Mount options: allow_other,nonempty,relatime,default_permissions,fsname=/dev/sda1,blkdev,blksize=4096
Nov  4 19:57:59 outernet daemon.notice ntfs-3g[1354]: Global ownership and permissions enforced, configuration type 1
Nov  4 19:57:59 outernet user.debug hotplug.sda1: Processing external storage
Nov  4 19:58:02 outernet user.debug hotplug.sda1: External storage platform is: (none)
Nov  4 19:58:02 outernet user.debug hotplug.sda1: Preparing for external storage use
Nov  4 19:58:02 outernet user.debug hotplug.sda1: Stopping services
Nov  4 19:58:02 outernet user.debug hotplug.sda1: Stoppnig S91librarian
Nov  4 19:58:02 outernet user.debug hotplug.sda1: Stoppnig S90fsal
Nov  4 19:58:02 outernet user.debug hotplug.sda1: Stoppnig S80ondd
Nov  4 19:58:02 outernet user.info kernel: [ 1132.430849@1] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 19:58:02 outernet user.info kernel: [ 1132.430873@1] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 19:58:02 outernet user.info kernel: [ 1132.430883@1] wetek-dvb dvb.8: FLUSH ok
Nov  4 19:58:05 outernet user.debug hotplug.sda1: Services stopped
Nov  4 19:58:05 outernet user.debug hotplug.sda1: Initializing external storage
Nov  4 19:58:05 outernet user.debug hotplug.sda1: Writing platform file to external storage
Nov  4 19:58:05 outernet user.debug hotplug.sda1: Copying content from internal storage to external
Nov  4 19:58:05 outernet user.debug hotplug.sda1: Copying /mnt/data/downloads/* -> /mnt/sda1
Nov  4 19:58:05 outernet user.warn hotplug.sda1: There were errors during copying
Nov  4 19:58:05 outernet user.info hotplug.sda1: Using /dev/sda1 as external storage
Nov  4 19:58:05 outernet daemon.notice ntfs-3g[1354]: Unmounting /dev/sda1 (EXT_STORE)
Nov  4 19:58:06 outernet user.debug hotplug.sda1: Restarting services
Nov  4 19:58:06 outernet daemon.notice ntfs-3g[1371]: Version 2015.3.14 integrated FUSE 27
Nov  4 19:58:06 outernet daemon.notice ntfs-3g[1371]: Mounted /dev/sda1 (Read-Write, label "EXT_STORE", NTFS 3.1)
Nov  4 19:58:06 outernet daemon.notice ntfs-3g[1371]: Cmdline options: windows_names,fmask=133,dmask=022,recover
Nov  4 19:58:06 outernet daemon.notice ntfs-3g[1371]: Mount options: allow_other,nonempty,relatime,default_permissions,fsname=/dev/sda1,blkdev,blksize=4096
Nov  4 19:58:06 outernet daemon.notice ntfs-3g[1371]: Global ownership and permissions enforced, configuration type 1
Nov  4 19:58:06 outernet user.info ondd[1376]: [main] v2.1.0
Nov  4 19:58:06 outernet user.info ondd[1376]: [main] tuning to 1179 Mhz 22000 KSym
Nov  4 19:58:06 outernet user.debug hotplug.sda1: Services started
```

(Note: NTFS microSD is not included here because the error is related to kernal I/O -  see #8 )
### Final note

On refresh files view only showed a .platform file, none of the content from /mnt/data/Downloads had been transferred.
## Removal behavior

On removal of FAT32/NTFS drives several things happen:
- FSAL crashes
- /mnt/data/downloads/ is unmounted
- as can be seen in the following logs, unmounting of /mnt/data/downloads/ is not logged.
#### NTFS - eHDD

```
Nov  4 20:09:48 outernet user.warn kernel: [ 1838.210005@0] Disconnect cb-Host
Nov  4 20:09:48 outernet user.info kernel: [ 1838.337607@1] usb 2-1: USB disconnect, device number 3
Nov  4 20:09:48 outernet user.debug hotplug.sda1: Stopping services
Nov  4 20:09:48 outernet user.debug hotplug.sda1: Stoppnig S91librarian
Nov  4 20:09:48 outernet user.debug hotplug.sda1: Stoppnig S90fsal
Nov  4 20:09:48 outernet user.debug hotplug.sda1: Stoppnig S80ondd
Nov  4 20:09:49 outernet user.info kernel: [ 1838.748899@0] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 20:09:49 outernet user.info kernel: [ 1838.748919@0] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 20:09:49 outernet user.info kernel: [ 1838.748928@0] wetek-dvb dvb.8: FLUSH ok
Nov  4 20:09:52 outernet user.debug hotplug.sda1: Services stopped
Nov  4 20:09:52 outernet daemon.notice ntfs-3g[1371]: Unmounting /dev/sda1 (EXT_STORE)
Nov  4 20:09:52 outernet user.debug hotplug.sda1: Restarting services
Nov  4 20:09:52 outernet user.info ondd[1425]: [main] v2.1.0
Nov  4 20:09:52 outernet user.debug hotplug.sda1: Services started
```

(Note: NTFS microSD is not included here because the error is related to kernal I/O -  see #8 )
### Final notes:
- On refresh of the files view no files are listed, though files are visible as expected with `ls` in /mnt/data/downloads/
- Reboot is required to restore correct folders.

**Comment by foxbunny** [comment]
Please report separately on FAT32 and NTFS. Filemanager showing "Go to complete file list" is a separate issue.

**Comment by ben-en** [comment]
Bug reports split as requested, see #9

**Comment by foxbunny** [comment]
This should be fixed with Outernet-Project/outernetrx-buildroot@d2bd555

**Comment by ben-en** [comment]
Resolved as of 08149db471c0eb545632bde2d09ba9b3010e578a
